feat: add accessor protocol and refactor `stats/base/nanvariancepn`

rahulptl165 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5675 .

## Description

This pull request:

-   Adds new file `lib/accessors.js`.
-   Modified `lib/nansumpw.js`.
-   Modified `lib/nanvariancepn.js`.
-   Modified `lib/ndarray.js`.
-   Modified examples, benchmarks, docs, tests and readme files accordingly.  


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5675 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

rahulptl165 committed:
feat: add accessor protocol and refactor stats/base/nanvariancepn

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte committed:
Merge branch 'develop' into feat/accessor-nanvariancepn

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/README.md:
```diff
@@ -98,7 +98,7 @@ The use of the term `n-1` is commonly referred to as Bessel's correction. Note,
-#### nanvariancepn( N, correction, x, stride )
+#### nanvariancepn( N, correction, x, strideX )
 
 Computes the [variance][variance] of a strided array `x` ignoring `NaN` values and using a two-pass algorithm.
```
```suggestion
Computes the [variance][variance] of a strided array ignoring `NaN` values and using a two-pass algorithm.
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/README.md:
```diff
@@ -134,41 +131,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
 var Float64Array = require( '@stdlib/array/float64' );
-var floor = require( '@stdlib/math/base/special/floor' );
 
 var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN ] );
```
```suggestion
var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/README.md:
```diff
@@ -134,41 +131,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
-var N = floor( x0.length / 2 );
-
-var v = nanvariancepn( N, 1, x1, 2 );
+var v = nanvariancepn( 4, 1, x1, 2 );
```
```suggestion
var v = nanvariancepn( 5, 1, x1, 2 );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/README.md:
````diff
@@ -134,41 +131,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
 ```javascript
-var floor = require( '@stdlib/math/base/special/floor' );
-
 var x = [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ];
````
```suggestion
var x = [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ];
```

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/README.md:
```diff
@@ -134,41 +131,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
-var N = floor( x.length / 2 );
 
-var v = nanvariancepn.ndarray( N, 1, x, 2, 1 );
+var v = nanvariancepn.ndarray( 4, 1, x, 2, 1 );
```
```suggestion
var v = nanvariancepn.ndarray( 5, 1, x, 2, 1 );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/docs/repl.txt:
```diff
@@ -1,10 +1,10 @@
 
-    The `N` and `stride` parameters determine which elements in `x` are accessed
-    at runtime.
+    The `N` and stride parameters determine which elements in the stride arrays
```
```suggestion
    The `N` and stride parameters determine which elements in the strided array
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/docs/types/test.ts:
```diff
@@ -25,7 +26,7 @@ import nanvariancepn = require( './index' );
 	const x = new Float64Array( 10 );
 
-	nanvariancepn( x.length, 1, x, 1 ); // $ExpectType number
+	nanvariancepn( x.length, 1, new AccessorArray( x ), 1 ); // $ExpectType number
```
```suggestion
	nanvariancepn( x.length, 1, x, 1 ); // $ExpectType number
	nanvariancepn( x.length, 1, new AccessorArray( x ), 1 ); // $ExpectType number
```

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/docs/types/test.ts:
```diff
@@ -100,7 +101,7 @@ import nanvariancepn = require( './index' );
 	const x = new Float64Array( 10 );
 
-	nanvariancepn.ndarray( x.length, 1, x, 1, 0 ); // $ExpectType number
+	nanvariancepn.ndarray( x.length, 1, new AccessorArray( x ), 1, 0 ); // $ExpectType number
```
```suggestion
	nanvariancepn.ndarray( x.length, 1, x, 1, 0 ); // $ExpectType number
	nanvariancepn.ndarray( x.length, 1, new AccessorArray( x ), 1, 0 ); // $ExpectType number
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/accessor-nanvariancepn

gururaj1512 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/accessor-nanvariancepn

gururaj1512 committed:
docs: update README for clarity and formatting

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
chore: change path

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
refactor: rename variable for clarity and remove unused file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
test: add accessors tests for nanvariancepn function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
docs: update docs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/nanvariancepn/lib/accessors.js:
```diff
@@ -44,17 +44,25 @@ var WORKSPACE = [ 0.0, 0 ];
 *
 * @param {PositiveInteger} N - number of indexed elements
```
```suggestion
* @private
* @param {PositiveInteger} N - number of indexed elements
```

kgryte committed:
docs: add missing private annotation

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
